app/utilities/proxyAgent/index.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+import SocksProxyAgent from 'socks-proxy-agent'
+import HttpProxyAgent from 'http-proxy-agent'
+
+import { remote } from 'electron'
+const logger = remote.getGlobal('logger')
+
+/**
+ * A socks4/socks5/http/https proxy agent
+ */
+export default function () {
+  // TODO: move to global configuration
```

Comment:
This can be done after resolve #86 (setting panel & global configuration mechanism).